### PR TITLE
Implement comprehensive query wildcard re-write

### PR DESCRIPTION
**Pull request by otoolep**
This patch implements comprehensive rewrite of wildcards, for both the "source" of the `SELECT` statement, and the `GROUP BY`. The parser prevents statements such as `SELECT *,value`, so those unit tests are commented out. However, the query parser should be fixed up eventually to allow this, as other SQL-based query engines support this.

### CODE REVIEW

**Comment by otoolep** [comment]
This changes uses dedicated code for `SELECT` query rewrite, and is very close in form to what existed already for `SELECT *`. It's not clear to me if we should be using some node-walking re-write code.

@pauldix @benbjohnson @corylanou

**Comment by otoolep** [comment]
Addresses issue #1624.

**Comment by otoolep** [comment]
@dgnorton -- could do with your review too.

File: server.go [hunk 1]
```diff
@@ -1977,6 +1977,12 @@ func (s *Server) ExecuteQuery(q *influxql.Query, database string, user *User) Re
 
 // executeSelectStatement plans and executes a select statement against a database.
 func (s *Server) executeSelectStatement(stmt *influxql.SelectStatement, database string, user *User) *Result {
+	// Perform any necessary query re-writing.
+	stmt, err := s.rewriteSelectStatement(stmt)
```
Comment (pauldix) [inline server.go]: We should only rewrite if the statement actually has a wildcard. The statement struct should have a method that indicates if it has a wildcard.

**Comment by corylanou** [comment]
+1

File: server.go (hunk 1 above)
Comment (benbjohnson) [inline server.go]: :+1:

**Comment by benbjohnson** [comment]
Are we going to handle wildcard expansion for multiple source measurements? e.g.

**Comment by otoolep** [comment]
Wildcard method added @pauldix @benbjohnson -- there are a couple of different places the check could be placed, of course.

File: influxql/parser.go [hunk 2]
```diff
@@ -1508,6 +1508,8 @@ func (p *Parser) parseUnaryExpr() (Expr, error) {
 	case DURATION_VAL:
 		v, _ := ParseDuration(lit)
 		return &DurationLiteral{Val: v}, nil
+	case MUL:
+		return &Wildcard{}, nil
```
Comment (dgnorton) [inline influxql/parser.go]: Wondering if wildcard expressions should be parsed separately so we don't allow weird stuff like `SELECT value + *,* FROM cpu`, which will get expanded to look something like `SELECT value + *, value1, value2 FROM cpu`.  /cc @benbjohnson

Comment (otoolep) [inline influxql/parser.go]: Sounds like a reasonable suggestion, but outside the scope of this change I'd say. Happy to open an issue.

**Comment by otoolep** [comment]
@pauldix is refactoring a chunk of the query engine, so this PR is on hold. He and @benbjohnson have promised to merge (or possibly rework) this when ready.

File: influxql/parser.go (hunk 2 above)
Comment (benbjohnson) [inline influxql/parser.go]: I agree that we should do a lookahead for wildcards before parsing expressions but we can push that to a separate PR.

**Comment by otoolep** [comment]
Rebased on master, and green build now.

**Pull request merged**